### ISSUE TITLE
Matching Cards Super Stuck

### ISSUE THREAD

**Issue by pdavies88** [ga-dc/project1#94]
I am having trouble identifying the matching part of my concentration game. I was able to shuffle the cards randomly and add a random image and value in my shuffled array to the cards. I am having the cards flip by adding a class to them on click that shows the image on the back of their div class. Each card is in its own div-container with two divs inside of it a div front that contains the front image and a div back that contains the back image. I am using $(this).val() to pull the value for the cards however I am not sure how to get them to match with the other card in the shuffled array that has the same value.  Also I believe that adding the new class on click is not allowing the $(this).val() to associate properly. Really hoping to not have to scrap my code and start from scratch but can if needed. Below is the JS code I have so far and have been stuck with for quite awhile:
As always thanks in advance

``` js
$(document).ready(function(){
  var numFlipped = 0;
    var flipCard = $('div').on("click",function(){
      $(this).addClass('flip')
            pickedCard = $(this).find("img.backCard");
            console.log(pickedCard);
            CardValue = $(this).val();
            console.log(CardValue)
          pickedCard.attr("src", imageLibrary[CardValue])
          numFlipped++;
          cardMatching();
        })

function cardMatching(){
  if (CardValue === $("div.flip-container.flip").val())
  console.log("match made")
}
    var cardDeck = [1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12];

    function shuffle(cardDeck){
      for(var j, x, i = cardDeck.length; i; j = Math.floor(Math.random() * i), x = cardDeck[--i], cardDeck[i] = cardDeck[j], cardDeck[j] = x);
      console.log(cardDeck);
    }
    shuffle(cardDeck);
    for (i = 0; i < cardDeck.length; i++) {
      $(".flip-container").eq(i).val(cardDeck[i]);
    }
    var imageLibrary = {
      1: "images/ermac.png",
      2: "images/jax.png",
      3: "images/johnny_cage.png",
      4: "images/kano.png",
      5: "images/kenshi.png",
      6: "images/liu_kang.png",
      7: "images/quan_chi.png",
      8: "images/radien.png",
      9: "images/reptile.png",
      10: "images/scorpion.png",
      11: "images/sonya_blade.png",
      12: "images/sub_zero.png"
    } 
});
```

**Comment by andrewsunglaekim** [comment]
What i would do, is any time you flip a card, you also push it to an array. That way you have an ordered list you can compare too.. heres the more like code/pseudo code im thinking of :

```
// click 
  // when i click 
    // if the counter is even i move the dom element into an array
    // if the counter is odd i move the dom element into an array, check against the last element in the array
      // if its the same card i leave all cards exposed and move the current element into the array
      // if its a different card i turn cards back over 
```

or something like that.

**Comment by jshawl** [comment]
I'm having trouble understanding which part of your code isn't working. can you please narrow down your issue, like:

When I click on a card, `this.val()` is undefined.

**Comment by pdavies88** [comment]
console log Shuffle Array - [12, 3, 5, 6, 9, 3, 7, 11, 4, 9, 5, 4, 10, 10, 11, 1, 12, 2, 1, 8, 6, 8, 2, 7]
console.log(pickedCard) - script.js:7 [prevObject: n.fn.init[1], context: div.front.flip, selector: "img.backCard"]
console.log(CardValue) - script.js:9 "this line is blank and counts as the first click which is shown below"
console.log(numFlipped) - script.js:12 1 - this is showing the first click
console.log(pickedCard) - script.js:7 [img.backCard, prevObject: n.fn.init[1], context: div.flip-container.flip, selector: "img.backCard"]
console.log(CardValue)  - script.js:9 12 - "now the value of the card from the array is input"
console.log(numFlipped) - script.js:12 2 - this is showing the second click
However both of these clicks are happening on the first click of the card
